### DISCUSSION TITLE
Running Holos Calculations from Python & Batch Processing for Multiple Farms

### DISCUSSION THREAD

**Discussion by CalvinSYLai** [holos-aafc/Holos#57] [General]
Hello everyone 👋,

I'm currently working on a project to quantify our GHG emissions for a large number of farms with sparse data and I have a few questions for anyone who has worked on the leveraging the Holos tool for specialized projects.

Python Integration:
Has anyone tried running Holos calculations directly from a Python codebase? We're interested in incorporating Holos' calculations, code, and parameter settings into our existing Python environment. If you have a recommended approach or example projects that would serve as a good reference, that would be extremely helpful.

Batch Processing:
Additionally, we're curious to know if anyone has experience running batch or table inputs with the Holos engine. Specifically, we're looking to feed multiple farm data at once and receive carbon output calculations in return. Any guidance on this would be invaluable.

Calvin Lai

**Comment by aaron-mcpherson** [comment]
Hi Calvin,

There is a command line version of the Holos application. The CLI can be used to run batches of farms through the engine.

There is a section in the user guide on how to get started with the CLI:

https://github.com/holos-aafc/Holos/blob/main/H.Content/Documentation/User%20Guide/User%20Guide.md#chapter-10---command-line-interface

You can run the CLI by compiling the source code or you can download an installer package for the CLI here:

https://github.com/holos-aafc/Holos/tree/main#download

Let me know if you have any other questions

Aaron

**Reply by ramsaym** [reply]
I can confirm I am able to run batch jobs using the GUI as a farm config tool and then exporting multiple farms to .json export for altering, filtering, processing via CLI
